Frameworks/CoreText/CTTypesetter.mm

```diff
@@ -191,32 +205,46 @@ static CFIndex _DoWrap(CTTypesetterRef ts, CFRange range, WidthFinderFunc widthF
             if (lastPossibleBreakPos != -1) {
                 //  We must now perform a soft break
                 curIndex = lastPossibleBreakPos + 1;
-
-                if (curChar == ' ') {
-                    //  Do a hard line break
-                    hitLinebreak = true;
-                }
                 lineWidth = ((float)lastPossibleBreakWidth) / 64.0f;
             } else {
                 if (lineStart != curIndex) {
                     //  Back out the last character
                     penX -= advance.x;
+                    lastGlyphToPrintPos = curIndex - 1;
+                } else {
+                    lastGlyphToPrintPos = curIndex;
                 }
                 lineWidth = ((float)penX) / 64.0f;
-                break;
             }
+
+            //  Do a hard line break
+            hitLinebreak = true;
             break;
         }
 
-        curIndex++;
+        ++curIndex;
     }
 
-    if (curIndex >= count) {
-        lineWidth = ((float)penX) / 64.0f;
+    if (curIndex >= count && !hitLinebreak) {
+        // Ran out of characters to print before establishing width
+        if (chars[count - 1] == ' ') {
+            // Ending line with trailing spaces, only want width up to last printable glyph
+            lineWidth = ((float)lastPossibleBreakWidth) / 64.0f;
```

Comment:
Since I imagine we'll rewrite most of this when switching it over to DWrite, I'll leave it as is

---
In reply to: [74542897](https://github.com/Microsoft/WinObjC/pull/873#discussion_r74542897) [](ancestors = 74542897,74528837)